amqtt/broker.py

```diff
@@ -421,15 +421,24 @@ async def client_connected(
             )
             # await writer.close()
             self.logger.debug("Connection closed")
+            server.release_connection()
             return
         except MQTTException as me:
             self.logger.error(
                 "Invalid connection from %s : %s"
                 % (format_client_message(address=remote_address, port=remote_port), me)
             )
             await writer.close()
+            server.release_connection()
             self.logger.debug("Connection closed")
             return
+        except NoDataException as ne:
+            self.logger.error(
+                "No data from %s : %s"
+                % (format_client_message(address=remote_address, port=remote_port), ne)
+            )
+            server.release_connection()
```

Comment:
Instead of calling `server.release_connection()` in every `except` block, maybe rather add a `finally` block for that

Reply:
Thanks for the feedback. `server.release_connection()` is not called in the `try` case. Thus, moving the call to `finally` will not be the same behaviour.

Reply:
Ah true, my bad :+1: